### PR TITLE
Adding device factory

### PR DESCRIPTION
**Pull request by PaulWieland**
`factory()` method added to InsteonDevice class

`queryDeviceInfo(...)` method added to PowerLincModem class (needed for the device factory method)

`toAddressArray(...)` added to utils so we can go between address strings and arrays in both directions consistently

There is a problem I haven't been able to solve yet with importing `KeypadDimmer` & `SwitchedLightingDevice` classes in `InsteonDevice.ts`. In both cases I get a `TypeError: Class extends value undefined is not a constructor or null`. The `DimmableLightingDevice` imports just fine.

### CODE REVIEW

**Comment by PaulWieland** [comment]
Here's a test script:

```javascript
const INSTEON = require('insteon-plm');
const PowerLincModem = INSTEON.default;
const InsteonDevice = INSTEON.InsteonDevice;
const DimmableLightingDevice = INSTEON.DimmableLightingDevice;
const SwitchedLightingDevice = INSTEON.SwitchedLightingDevice;

async function main(){

  /* Setup */
  const modem = new PowerLincModem('/dev/ttyPLM');

  modem.on('ready', async () => {

    console.log('Modem Ready');

	/* This device is a dimmer switch */
	const dining = [0x31,0x50,0x35];
	
	let myDevice = await InsteonDevice.factory(dining, modem, {});
	console.log("constructor",myDevice.constructor.name);
  });

}

main();

/* Outputs
Modem Ready
constructor DimmableLightingDevice
*/

```

File: src/PowerLincModem.ts [hunk 1]
```diff
@@ -746,6 +746,30 @@ export default class PowerLincModem extends EventEmitter2{
...
 	//#endregion
+	
+	/* Send an insteon command from the modem to a device to find out what it is */
+	public queryDeviceInfo = (deviceID: Byte[]) => new Promise<Packet.StandardMessageRecieved>((resolve, reject) => {
+
+		// Catching broadcast message
+		this.once(
+			['p', PacketID.StandardMessageReceived.toString(16), MessageSubtype.BroadcastMessage.toString(16), toAddressString(deviceID)],
+			(data: Packet.StandardMessageRecieved) => {
+				if(!Array.isArray(data.to)) reject(data);
```
Comment (apearson) [inline src/PowerLincModem.ts:757]: Not sure why this is in here

File: src/utils.ts [hunk 2]
```diff
@@ -9,6 +9,10 @@ export function toAddressString(address: Byte[]){
   return address.map(num => num.toString(16).toUpperCase()).join('.');
 }
 
+export function toAddressArray(address: String){
+	return address.split(".").map(el => parseInt(el,16));
+}
+
```
Comment (apearson) [inline src/utils.ts:15]: This is much needed, thanks

**Review by apearson — APPROVED** [review]

**Comment by apearson** [comment]
Thanks!

File: src/PowerLincModem.ts (hunk 1 above)
Comment (PaulWieland) [reply src/PowerLincModem.ts:757]: It's not needed. Wasn't sure at the time if `to` was always an array.

**Pull request merged**